### ISSUE TITLE
Downloads unsuccessful when trying to download apk from app

### ISSUE THREAD

**Issue by rancidfrog** [thermatk/FastHub#66]
**FastHub-Libre Version: 4.6.7**  
**Android Version: 7.1.1 (SDK: 25)**  
**Device Information:**  
- **Manufacturer:** OnePlus  
- **Brand:** OnePlus  
- **Model:** OnePlus5  
---

 Whenever I try to download apk from release sections, I get download unsuccessful from download manager, forcing me to open with browser then download.
Do not know where the problem lies, as app has storage permission and besides it just sends request to download manager anyway

**Comment by RebelLion420** [comment]
I have the same exact issue when trying to download apks, I was trying to get Material Glass Substratum apk. Fails every time with every release, even with tar.gz files. Only zips seem to download no issue

Device: Moto E4 (perry)
ROM: crDroid 7.1.2 unofficial

**Comment by 0x5ECF4ULT** [comment]
Same issue. It's getting kind of annoying. I really love this app but it lacks of this specific feature. Can the developer (@thermatk) be so kind and take care of this issue? I think this is an essential feature if someone wants to download the sources in another MIME type than .zip or even the compiled binary! 

Device: SM-G930F
OS: havocOS (LineageOS) 8.1.0

**Comment by walrus543** [comment]
Same thing but with a log.
OnePlus 3T - OmniDragon Oreo 8.1 - Fasthub Libre 4.6.7 downloaded from F-Droid

`Path appears to be invalid: /storage/emulated/0/FastHub-Libre/NetGuard-v2.207-release.apk`
This path does exists.

[logcat_09-12-2018_10-26-58.txt](https://github.com/thermatk/FastHub-Libre/files/2374214/logcat_09-12-2018_10-26-58.txt)

**Comment by 0x5ECF4ULT** [comment]
@Primokorn It is obvious that the download manager should be redesigned. I will fork this repo and investigate this issue + PR

**Comment by rancidfrog** [comment]
But it used to work fine.
And folder, FastHub-Libre, exists and has all apks that were downloaded via app before it broke all of a sudden

**Comment by 0x5ECF4ULT** [comment]
@rancidfrog yeah that's true. I just took a look into the Fasthub folder... This is really strange. I'll take a look at the source code which is responsible for the DM

**Comment by repsorp** [comment]
Same here

Android 8.1
FastHub-Libre 4.6.7 

> DownloadManager(19610): Path appears to be invalid: /storage/emulated/0/FastHub-Libre/QKSMS-v3.5.4.apk 

![whatsapp image 2018-09-21 at 18 37 40](https://user-images.githubusercontent.com/6297032/45894968-44356f80-bdd0-11e8-8187-3da0ae5b3024.jpeg)

**Comment by 0x5ECF4ULT** [comment]
@repsorp maybe because of the owner. I guess the DM fails because the owner is root🤔 try to change the owner (you will need root permissions). I wonder if it still fails after this change

**Comment by repsorp** [comment]
@tacticalDevC 

Couldn't change the owner.

I'm using Magisk with which all repertories are owned by root who appears to be the regular owner.

Permissions are ok though, rwx/770 for both group and owner

**Comment by SecT0uch** [comment]
Same problem here. Any update?

**Comment by repsorp** [comment]
@SecT0uch nope...
Not any update regarding this issue

**Comment by thermatk** [comment]
Very sorry, but nothing I can do till v5 is finalised upstream. They abandoned v4 for now

**Comment by repsorp** [comment]
@thermatk don't be sorry bro, you already made such a good job.
Thanks for everything and keep up the good work :v:
